app/controllers/trainer_pokemons_controller.rb

```diff
@@ -1,6 +1,32 @@
 class TrainerPokemonsController < ApplicationController
   def index
     @trainer = Trainer.find(params[:trainer_id])
-    @pokemons = @trainer.pokemons
+    @pokemons = @trainer.pokemons.order(params[:order] || params[:id])
```

Comment:
was hard to figure out how to apply query params to make this default order by id change only after clicking link to order by name